playbooks/operationalizing/vault_integration.adoc

```diff
@@ -0,0 +1,169 @@
+----
+=== Unseal Vault.
+
+You have to repeat this step every time you start vault. 
+Don't try to automate this step, this is manual by design. 
+You can make the initial seal stronger by increasing the number of keys. 
+We will assume that the KEYS environment variable contains the key necessary to unseal the vault and that ROOT_TOKEN contains the root token.
+
+For example: 
+----
+export KEYS=9++8KEDd72S3aGc0zaY9JW11tnQRDTEkCZWMHK2D0CM= 
+export ROOT_TOKEN=c30909da-a713-94bf-bf6e-46180ef79a64 
+vault unseal -tls-skip-verify $KEYS
+----
+
+=== Install Vault Controller
+----
+oc create secret generic vault-controller --from-literal vault-token=$ROOT_TOKEN
+oc adm policy add-cluster-role-to-user view system:serviceaccount:vault-controller:default
+oc create -f ./openshift/vault-controller.yaml
+----
+== Running the example
+
+At this point you are ready to start deploying pods that use this approach to inject secrets. 
+The repository comes with two example one for a https://github.com/raffaelespazzoli/credscontroller/tree/master/examples/spring-example[Vault aware app] and one for a https://github.com/raffaelespazzoli/credscontroller/tree/master/examples/spring-legacy-example[legacy app].  
+Here are the instructions to run the vault aware example. 
+In this first example a https://projects.spring.io/spring-boot/[spring boot] app uses the spring https://github.com/spring-cloud/spring-cloud-vault[cloud vault configuration plugin] to get part of its configuration from Vault. 
+The Init Container will write the unwrapped Vault token to a well know location, the app will use that token to authenticate with Vault and retrieve its credentials.
+
+=== Create the Vault policy
+
+We need to create a policy that will allow to access a specific subset of secrets. This is how you can have multiple application storing their secrets in Vault.
+----
+export VAULT_TOKEN=$ROOT_TOKEN
+vault policy-write -tls-skip-verify spring-example ./examples/spring-example/spring-example.hcl
+----
+=== Create the secret
+
+We now store a static secret in Vault +
+----
+vault write -tls-skip-verify secret/spring-example password=pwd
+----
+=== Build the app
+----
+oc new-project spring-example
+oc new-build registry.access.redhat.com/redhat-openjdk-18/openjdk18-openshift~https://github.com/raffaelespazzoli/credscontroller --context-dir=examples/spring-example --name spring-example
+----
+=== Allow the app to talk to Vault and Vault Controller
+
+You need to execute this step only if you are running in a multi-tenant OpenShift 
+----
+oc adm pod-network join-projects --to vault-controller spring-example
+----
+=== Deploy the app
+----
+oc create -f ./examples/spring-example/spring-example.yaml
+oc expose svc spring-example
+----
+=== Verify the app
+----
+export SPRING_EXAMPLE_ADDR=http://`oc get route | grep -m1 spring | awk '{print $2}'`
+curl $SPRING_EXAMPLE_ADDR/secret 
+----
+The last command should return the secret that was initially stored in Vault.
+
+
```

Comment:
Remove extra lines.

This guide should also have a conclusion/summary section to wrap things up and any links to reference additional resources (if any).